### DISCUSSION TITLE
Include the endpoint URI when generating documentation for a route handler

### DISCUSSION THREAD

**Discussion by jonathanGB** [rwf2/Rocket#2318] [Questions]
I have a binary crate with a Rocket server. I have thoroughly documented the server so that my partner working on the front-end can easily understand what the API is.

As it goes in Rust, I run `cargo doc`, but one thing I have noticed is that the generated HTML does not mention the URI endpoints (see below). For instance, if I have:

```rust
#[put(
    "/game/<_>/player/color",
    format = "json",
    data = "<change_color_request>"
)]
pub fn change_player_color(...) { ... }
```

It would be nice for the screenshot below to mention that the handler is attached to `PUT` requests going to `/game/<_>/player/color`.

![image](https://user-images.githubusercontent.com/8869660/184557617-b1473b46-a520-4801-b3a7-9c415a0fc027.png)

Of course, I could manually write the URI endpoint in the doc comment, but then the comment could become obsolete if I change the actual route, but forget to update the doc comment. Does anyone know about a good way to circumvent this issue? Would it be possible to tell the [route macro](https://api.rocket.rs/v0.5-rc/rocket/attr.route.html) to generate such documentation?

**Comment by the10thWiz** [comment]
This is a little complicated, but `https://github.com/GREsau/okapi` likely already implements the feature you want. It generates Swagger docs for the web API, but this is likely more readable for interacting with the Web API than Rust doc anyway. I believe okapi also handles mount points as well, so Swagger will actually include the mount point and so forth.
